src/hangar/repository.py

```diff
@@ -464,33 +481,158 @@ def create_branch(self, name: str, base_commit: str = None) -> str:
 
         Returns
         -------
-        str
-            name of the branch which was created
+        heads.BranchHead
+            NamedTuple[str, str] with fields for `name` and `digest` of the branch
+            created (if the operation was successful)
+
+        Raises
+        ------
+        ValueError
+            If the branch name provided contains characters outside of alpha-numeric
+            ascii characters and ".", "_", "-" (no whitespace).
+        ValueError
+            If the branch already exists.
+        RuntimeError
+            If the repository does not have at-least one commit on the `default`
+            (ie. `master`) branch.
         """
         self.__verify_repo_initialized()
-        if not is_suitable_user_key(name):
-            e = ValueError(
-                f'Branch name provided: `{name}` invalid. Must only contain '
-                f'alpha-numeric or "." "_" "-" ascii characters.')
+        if (not is_ascii(name)) or (not is_suitable_user_key(name)):
```

Comment:
`is_suitable_user_key` has created for `ascii` check? Perhaps merge both?

Reply:
`is_suitable_user_key` checks for either `str` (ascii) or `int` keys for use in checking sample names. `is_ascii` requires `str` typed data (for `arrayset` names, or this...) 

I like the effort to catch this and reduce dedundency, but they apply to different use cases.